src/OSPSuite.Infrastructure.Import/Core/DataSheetCollection.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using OSPSuite.Utility.Collections;
+
+namespace OSPSuite.Infrastructure.Import.Core
+{
+   /// <summary>
+   ///    a collection of excel dataSheet sheets
+   /// </summary>
+   public class DataSheetCollection : IEnumerable<DataSheet>
+   {
+      private readonly Cache<string, DataSheet> _dataSheets;
+
+      public DataSheetCollection()
+      {
+         _dataSheets = new Cache<string, DataSheet>(x => x.SheetName);
+      }
+
+      public void AddSheet(DataSheet sheet)
+      {
+         _dataSheets.Add(sheet);
+      }
+
+      public DataSheet GetDataSheet(string sheetName)
+      {
+         return _dataSheets[sheetName];
+      }
+
+      public DataSheetCollection GetDataSheets(IReadOnlyList<string> sheetNames)
```

Comment:
GetOrCreate. This has side effects and is not expected. Why is this needed?

Reply:
@msevestre again, maybe I am missing something here....let's talk about this